<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
    
    <head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
        <!-- AppResources meta begin -->
        <script type="text/javascript">var ncbi_startTime = new Date();</script>
        <!-- AppResources meta end -->
        
        <!-- TemplateResources meta begin -->
        <meta name="paf_template" content="" />

        <!-- TemplateResources meta end -->
        
        <!-- Logger begin -->
        <meta name="ncbi_db" content="pmc" /><meta name="ncbi_pdid" content="article" /><meta name="ncbi_acc" content="" /><meta name="ncbi_domain" content="procamia" /><meta name="ncbi_report" content="record" /><meta name="ncbi_type" content="fulltext" /><meta name="ncbi_objectid" content="" /><meta name="ncbi_pcid" content="/articles/PMC2655800/" /><meta name="ncbi_app" content="pmc" />
        <!-- Logger end -->
        
        <title>An Unsupervised Machine Learning Approach to Segmentation of Clinician-Entered Free Text</title>
        
        <!-- AppResources external_resources begin -->
        <link rel="stylesheet" href="/core/jig/1.14.8/css/jig.min.css" /><script type="text/javascript" src="/core/jig/1.14.8/js/jig.min.js"></script>

        <!-- AppResources external_resources end -->
        
        <!-- Page meta begin -->
        <meta name="robots" content="INDEX,NOFOLLOW,NOARCHIVE" /><link rel="canonical" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2655800/" /><link rel="schema.DC" href="http://purl.org/DC/elements/1.0/" /><meta name="citation_journal_title" content="AMIA Annual Symposium Proceedings" /><meta name="citation_title" content="An Unsupervised Machine Learning Approach to Segmentation of Clinician-Entered Free Text" /><meta name="citation_authors" content="Jesse O. Wrenn, Peter D. Stetson, Stephen B. Johnson" /><meta name="citation_date" content="2007" /><meta name="citation_volume" content="2007" /><meta name="citation_firstpage" content="811" /><meta name="citation_abstract_html_url" content="/pmc/articles/PMC2655800/?report=abstract" /><meta name="citation_pmid" content="18693949" /><meta name="DC.Title" content="An Unsupervised Machine Learning Approach to Segmentation of Clinician-Entered Free Text" /><meta name="DC.Type" content="Text" /><meta name="DC.Publisher" content="American Medical Informatics Association" /><meta name="DC.Contributor" content="Jesse O. Wrenn" /><meta name="DC.Contributor" content="Peter D. Stetson" /><meta name="DC.Contributor" content="Stephen B. Johnson" /><meta name="DC.Date" content="2007" /><meta name="DC.Language" content="en" /><meta property="og:title" content="An Unsupervised Machine Learning Approach to Segmentation of Clinician-Entered Free Text" /><meta property="og:type" content="article" /><meta property="og:description" content="Natural language processing is an important tool in biomedicine, and fails without successful segmentation of words and sentences. Tokenization is a form of segmentation that identifies boundaries separating semantic units, for example words, dates, numbers ..." /><meta property="og:url" content="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2655800/" /><meta property="og:site_name" content="PubMed Central (PMC)" /><meta property="og:image" content="https://www.ncbi.nlm.nih.gov/corehtml/pmc/pmcgifs/pmc-logo-share.png" /><meta name="twitter:card" content="summary" /><meta name="twitter:site" content="@ncbi" /><link rel="stylesheet" href="/corehtml/pmc/css/3.7/pmc.min.css" type="text/css" /><link rel="stylesheet" href="/corehtml/pmc/css/3.7/pmc_extras_prnt.min.css" type="text/css" media="print" /><script type="text/javascript" src="/corehtml/pmc/js/common.min.js">//</script><script type="text/javascript" src="/corehtml/pmc/js/NcbiTagServer.min.js">//</script><meta name="citationexporter" content="backend:'/pmc/utils/ctxp/'" /><script type="text/javascript" src="/corehtml/pmc/ctxp/jquery.citationexporter.js">//</script><link rel="stylesheet" href="/corehtml/pmc/ctxp/citationexporter.css" type="text/css" /><script type="text/javascript" src="/core/mathjax/2.6.1/MathJax.js?config=/corehtml/pmc/js/mathjax-config-classic.3.4.js"></script><script type="text/javascript" src="/corehtml/pmc/js/large-obj-scrollbars.min.js"></script><script type="text/javascript">window.name="mainwindow";</script><style type="text/css">.pmc-wm {background:transparent repeat-y top left;background-image:url(/corehtml/pmc/pmcgifs/wm-procamia.gif);background-size: auto, contain}</style><style type="text/css">.print-view{display:block}</style><style type="text/css">
        div.pmc_para_cit li.highlight,
        div.pmc_para_cit li.highlight .one_line_source
        { background: #E0E0E0; }
        a.bibr.highlight { background: #E0E0E0; } 
      </style><meta name="cited_in_systematic_reviews" content="" /><link rel="alternate" type="application/epub+zip" href="/pmc/articles/PMC2655800/epub/" /><link rel="alternate" type="application/pdf" href="/pmc/articles/PMC2655800/pdf/amia-0811-s2007.pdf" />

        <!-- Page meta end -->
    <link rel="shortcut icon" href="//www.ncbi.nlm.nih.gov/favicon.ico" /><meta name="ncbi_phid" content="F4FC06DA94E13C510000000000E200B8" />
<meta name='referrer' content='origin-when-cross-origin'/><link type="text/css" rel="stylesheet" href="//static.pubmed.gov/portal/portal3rc.fcgi/4158155/css/3852956/3985586/3808861/4121862/3974050/3917732/251717/4048120/3846471/14534/45193/4113719/3849091/3984811/3751656/4033350/3840896/3577051/3852958/3881636/3579733/4062871/12930/3964959/3855473/4047625/3854974/3854955/4076335/4128070/9685/3549676/3609192/3609193/3609213/3395586/4143404.css" /><link type="text/css" rel="stylesheet" href="//static.pubmed.gov/portal/portal3rc.fcgi/4158155/css/3411343/3882866/4157116.css" media="print" /></head>
    <body class="article">
        <div class="grid">
            <div class="col twelve_col nomargin shadow">
                <!-- System messages like service outage or JS required; this is handled by the TemplateResources portlet -->
                <div class="sysmessages">
                    <noscript>
	<p class="nojs">
	<strong>Warning:</strong>
	The NCBI web site requires JavaScript to function. 
	<a href="/guide/browsers/#enablejs" title="Learn how to enable JavaScript" target="_blank">more...</a>
	</p>
	</noscript>
                </div>
                <!--/.sysmessage-->
                <div class="wrap">
                    <div class="page">
                        <div class="top">
                            <div class="universal_header" id="universal_header"><ul class="inline_list jig-ncbimenu ui-ncbimenu resources_list" id="navcontent"><li class="ui-ncbimenu-item-leaf ui-ncbimenu-item-first ui-helper-reset ui-ncbimenu-item-no-hlt"><a class="ui-ncbimenu-link-first" href="/" role="banner" title="NCBI Home" id="ncbihome" accesskey="1"><span class="offscreen_noflow">NCBI</span><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4158155/img/28977" class="ncbi_logo" title="NCBI" alt="NCBI Logo" /></a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="#maincontent" title="Skip to the content" tabindex="0" accesskey="3">Skip to main
                        content</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="#navcontent" title="Skip to the navigation" tabindex="0" accesskey="4">Skip to
                        navigation</a></li><li id="resource-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first ui-helper-reset"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="/static/header_footer_ajax/submenu/#resources">Resources</a></li><li id="all-howtos-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="/static/header_footer_ajax/submenu/#howto">How To</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="/guide/browsers/#accesskeys" title="About My NCBI Accesskeys" tabindex="0" accesskey="0">About NCBI Accesskeys</a></li></ul><div class="myncbi"><span id="myncbiusername" style="display:none"><a href="/account/settings/" id="mnu" title="Edit account settings"></a></span><a accesskey="2" href="/myncbi/" id="myncbi" style="display:none">My NCBI</a><a href="/account/" id="sign_in">Sign in to NCBI</a><a href="/account/signout/" id="sign_out" style="display:none">Sign Out</a></div></div>
                            <div class="header">
    <div class="res_logo">
  <h1 class="img_logo"><a href="/pmc/" class="pmc_logo offscreen">PMC</a></h1>
  <div class="NLMLogo">
    <a href="https://www.nlm.nih.gov/" title="US National Library of Medicine">US National Library of Medicine</a>
    <br />
    <a href="https://www.nih.gov/" title="National Institutes of Health">National Institutes of Health</a>
  </div>
</div>
    <div class="search"><form method="get" action="/pmc/"><div class="search_form"><label for="database" class="offscreen_noflow">Search database</label><select id="database"><optgroup label="Recent"><option value="pmc" selected="selected" class="last" data-ac_dict="pmc-search-autocomplete">PMC</option></optgroup><optgroup label="All"><option value="gquery">All Databases</option><option value="assembly">Assembly</option><option value="biocollections">Biocollections</option><option value="bioproject">BioProject</option><option value="biosample">BioSample</option><option value="biosystems">BioSystems</option><option value="books">Books</option><option value="clinvar">ClinVar</option><option value="clone">Clone</option><option value="cdd">Conserved Domains</option><option value="gap">dbGaP</option><option value="dbvar">dbVar</option><option value="nucest">EST</option><option value="gene">Gene</option><option value="genome">Genome</option><option value="gds">GEO DataSets</option><option value="geoprofiles">GEO Profiles</option><option value="nucgss">GSS</option><option value="gtr">GTR</option><option value="homologene">HomoloGene</option><option value="medgen">MedGen</option><option value="mesh">MeSH</option><option value="ncbisearch">NCBI Web Site</option><option value="nlmcatalog">NLM Catalog</option><option value="nuccore">Nucleotide</option><option value="omim">OMIM</option><option value="pmc" data-ac_dict="pmc-search-autocomplete">PMC</option><option value="popset">PopSet</option><option value="probe">Probe</option><option value="protein">Protein</option><option value="proteinclusters">Protein Clusters</option><option value="pcassay">PubChem BioAssay</option><option value="pccompound">PubChem Compound</option><option value="pcsubstance">PubChem Substance</option><option value="pubmed">PubMed</option><option value="pubmedhealth">PubMed Health</option><option value="snp">SNP</option><option value="sparcle">Sparcle</option><option value="sra">SRA</option><option value="structure">Structure</option><option value="taxonomy">Taxonomy</option><option value="toolkit">ToolKit</option><option value="toolkitall">ToolKitAll</option><option value="toolkitbook">ToolKitBook</option><option value="toolkitbookgh">ToolKitBookgh</option><option value="unigene">UniGene</option></optgroup></select><div class="nowrap"><label for="term" class="offscreen_noflow" accesskey="/">Search term</label><div class="nowrap"><input type="text" name="term" id="term" title="Search PMC. Use up and down arrows to choose an item from the autocomplete." value="" class="jig-ncbiclearbutton jig-ncbiautocomplete" data-jigconfig="dictionary:'pmc-search-autocomplete',disableUrl:'NcbiSearchBarAutoComplCtrl'" autocomplete="off" data-sbconfig="ds:'no',pjs:'no',afs:'yes'" /></div><button id="search" type="submit" class="button_search nowrap" cmd="go">Search</button></div></div></form><ul class="searchlinks inline_list"><li>
                        <a href="/pmc/advanced/">Advanced</a>
                    </li><li>
                        <a href="/pmc/journals/">Journal list</a>
                    </li><li class="help">
                        <a target="_blank" href="/books/NBK3825/">Help</a>
                    </li></ul></div>
</div>

                            
                            
                        <!--<component id="Page" label="headcontent"/>-->
                            
                        </div>
                        <div class="content">
                            <!-- site messages -->
                            <div class="container">
    <div id="maincontent" class="content eight_col col">
        <div class="navlink-box">
            <ul class="page-breadcrumbs inline_list small"><li class="journal-list"><a href="/pmc/journals/" class="navlink">Journal List</a></li><li class="archive"><a class="navlink" href="/pmc/journals/362/">AMIA Annu Symp Proc</a></li><li class="issue-page"><a class="navlink" href="/pmc/issues/177326/">v.2007; 2007</a></li><li class="accid">PMC2655800</li></ul>
        </div>

        <!-- Journal banner -->
        <div class="pmc-page-banner whole_rhythm"><div><img src="/corehtml/pmc/pmcgifs/logo-procamia.gif" alt="Logo of procamia" usemap="#logo-imagemap" /><map id="logo-imagemap" name="logo-imagemap"><area alt="Link to Publisher's site" title="Link to Publisher's site" shape="default" coords="0,0,499,74" href="http://proceedings.amia.org/" target="pmc_ext" ref="reftype=publisher&amp;article-id=2655800&amp;issue-id=177326&amp;journal-id=362&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A&amp;rendering-type=normal" /></map></div> </div>
        
        <!--component id='MainPortlet' label='search-reference'/-->
        
        <!-- Book content -->
        <div class="">
            
        
            
            <div class="hide-overflow article lit-style content pmc-wm slang-all page-box"><!--main-content--><div class="jig-ncbiinpagenav" data-jigconfig="smoothScroll: false, allHeadingLevels: ['h2'], headingExclude: ':hidden'"><div class="fm-sec half_rhythm no_top_margin"><div class="fm-citation half_rhythm no_top_margin clearfix"><div class="small"><div class="inline_block eight_col va_top"><div><div><span class="cit"><span id="pmcmata">AMIA Annu Symp Proc</span>. 2007; 2007: 811–815. </span></div><div><span class="fm-vol-iss-date">Published online 2007. </span>  </div></div></div><div class="inline_block four_col va_top show-overflow align_right"><div class="fm-citation-ids"><div class="fm-citation-pmcid"><span class="fm-citation-ids-label">PMCID: </span><span>PMC2655800</span></div></div></div></div></div><h1 class="content-title">An Unsupervised Machine Learning Approach to Segmentation of Clinician-Entered Free Text</h1><div class="half_rhythm"><div class="contrib-group fm-author"><a href="/pubmed/?term=Wrenn%20JO%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=18693949">Jesse O. Wrenn</a>, BA,<sup>1</sup> <a href="/pubmed/?term=Stetson%20PD%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=18693949">Peter D. Stetson</a>, MD, MA,<sup>1,</sup><sup>2</sup> and  <a href="/pubmed/?term=Johnson%20SB%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=18693949">Stephen B. Johnson</a>, PhD<sup>1</sup></div></div><div class="fm-panel small half_rhythm"><div class="fm-authors-info fm-panel hide half_rhythm" id="idm140522521857440_ai" style="display:none"><div class="fm-affl" lang="en"><sup>1</sup>Department of Biomedical Informatics, Columbia University, New York NY</div><div class="fm-affl" lang="en"><sup>2</sup>Department of Medicine, Columbia University, New York, NY</div></div><div class="togglers"><a href="#" class="pmctoggle" rid="idm140522521857440_ai">Author information <span>►</span></a> <a href="#" class="pmctoggle" rid="idm140522521857440_cpl">Copyright and License information <span>►</span></a></div><div class="fm-article-notes fm-panel half_rhythm"></div><div class="fm-cpl-info fm-panel hide half_rhythm" id="idm140522521857440_cpl" style="display:none"><div class="fm-copyright half_rhythm"><a href="/pmc/about/copyright/">Copyright</a> &#x000a9;2007 AMIA - All rights reserved.</div><div class="fm-copyright half_rhythm">This is an Open Access article: verbatim copying and redistribution of this article are permitted in all media for any purpose</div></div></div><div id="pmclinksbox" class="links-box whole_rhythm"><div class="small"><div>This article has been <a href="/pmc/articles/PMC2655800/citedby/">cited by</a> other articles in PMC.</div></div></div></div><div class="sec"></div><div id="__abstractidm140522442194848" lang="en" class="tsec sec"><h2 class="head no_bottom_margin" id="__abstractidm140522442194848title">Abstract</h2><!--article-meta--><div><p id="__p1" class="p p-first-last">Natural language processing is an important tool in biomedicine, and fails without successful segmentation of words and sentences. Tokenization is a form of segmentation that identifies boundaries separating semantic units, for example words, dates, numbers and symbols, within a text. We sought to construct a highly generalizeable tokenization algorithm with no prior knowledge of characters or their function, based solely on the inherent statistical properties of token and sentence boundaries. Tokenizing clinician-entered free text, we achieved precision and recall of 92% and 93%, respectively compared to a whitespace token boundary detection algorithm. We classified over 80% of punctuation characters correctly, based on manual disambiguation with high inter-rater agreement (kappa = 0.916). Our algorithm effectively discovered properties of whitespace and punctuation in the corpus without prior knowledge of either. Given the dynamic nature of biomedical language, and the variety of distinct sublanguages used, the effectiveness and generalizability of our novel tokenization algorithm make it a valuable tool.</p></div></div><div id="__sec1" class="tsec sec"><h2 class="head no_bottom_margin" id="__sec1title">Introduction</h2><p id="__p2" class="p p-first">Natural Language Processing (NLP) has many applications in biomedicine<sup><a href="#b1-amia-0811-s2007" rid="b1-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_238636863">1</a></sup><sup>,</sup><sup><a href="#b2-amia-0811-s2007" rid="b2-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_238636865">2</a></sup>. Medical performance measurement is on the rise<sup><a href="#b3-amia-0811-s2007" rid="b3-amia-0811-s2007" class=" bibr popnode">3</a></sup>, and data about patient care, important in the assessment of quality<sup><a href="#b4-amia-0811-s2007" rid="b4-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_238636864">4</a></sup>, may be extracted from textual records. Online literature is another important resource in biomedicine<sup><a href="#b2-amia-0811-s2007" rid="b2-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_238636866">2</a></sup>, although there is now such a large quantity of literature that large-scale manual knowledge retrieval is impractical<sup><a href="#b5-amia-0811-s2007" rid="b5-amia-0811-s2007" class=" bibr popnode">5</a></sup>.</p><p id="__p3">NLP fails without successful segmentation<sub>,</sub> the process of separating text into processable units, such as words and sentences<sup><a href="#b6-amia-0811-s2007" rid="b6-amia-0811-s2007" class=" bibr popnode">6</a></sup>. The high granularity of language allows for segmentation on different levels. The most fine-grained semantic units of language are morphemes. Tokens are semantically independent combinations of morphemes, and sentences are combinations of tokens. Tokenization is the process of finding boundaries between tokens in a text<sup><a href="#b7-amia-0811-s2007" rid="b7-amia-0811-s2007" class=" bibr popnode">7</a></sup>.</p><p id="__p4">Intuitively, we could consider punctuation and whitespace as indications of token boundaries. However, the ambiguity of punctuation<sup><a href="#b8-amia-0811-s2007" rid="b8-amia-0811-s2007" class=" bibr popnode">8</a></sup>, especially in the medical language (<em>?rales</em>, <em>d/c</em>, <em>&#x02013;&#x0003e;</em>, <em>+/&#x02013;</em>, <em>0.05</em>), makes it difficult to use it to draw conclusions. Furthermore, while whitespace may frequently exist between tokens, some semantic units consist of two or more words (<em>diabetes mellitus</em>, <em>vital signs</em>, <em>S. aureus</em>), and some boundaries are not indicated by whitespace (<em>chest pain/palpitations/sob/diaphoresis/ nausea/vomiting</em>). Inference of token boundaries using whitespace alone is therefore frequently inaccurate.</p><p id="__p5">Most tokenizers are iteratively handcrafted on specific corpora. While such tokenizers may provide high corpus-specific recall and precision, they are very labor intensive to build and maintain, and rarely generalize well. Medicine has been shown to contain many distinct sublanguages with special properties<sup><a href="#b9-amia-0811-s2007" rid="b9-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_327591657">9</a></sup>. Text that has been entered directly by clinicians is highly and inconsistently abbreviated<sup><a href="#b10-amia-0811-s2007" rid="b10-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_327591656">10</a></sup>, arbitrarily capitalized and subject to dynamic changes. Therefore, the ability to quickly retrain a tokenization engine is integral to its long-term success. Examples of typing errors and arbitrarily punctuated, capitalized and abbreviated text segments from our set of clinician-entered text are as follows:
</p><blockquote class="pullquote"><p id="__p6"><em>1) stll c sob when walks &#x0003e; 1 b;lock tring do excrecises</em> [&#x02026;] <em>continue susing flovent an datrovent as rxed and trying do do inc execrcise</em> [sic]</p><p id="__p7">[&#x02026;] <em>R int. carotid art., eval. by neurosurg. for ? R CEA but carotid dopplers c/w LICA &#x0003c;35%</em> [sic]</p></blockquote><p></p><p id="__p8">Sentence boundary detection is also a complex problem. Clinician-entered free text frequently lacks sentence ending punctuation, and when it exists, it is often ambiguous<sup><a href="#b7-amia-0811-s2007" rid="b7-amia-0811-s2007" class=" bibr popnode">7</a></sup><sup>,</sup><sup><a href="#b8-amia-0811-s2007" rid="b8-amia-0811-s2007" class=" bibr popnode">8</a></sup>. Abbreviations, acronyms and contractions are very prominent in medicine<sup><a href="#b10-amia-0811-s2007" rid="b10-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_327591659">10</a></sup>, adding to the difficulty of punctuation disambiguation. Question marks end interrogative sentences, but also frequently precede questionable findings in medical text.</p><p id="__p9">We focused on generalizability and ease of maintenance as the most important features of a successful tokenization algorithm. Therefore we sought to construct an algorithm based on the inherent statistical properties of token boundaries. We hypothesized that changes in transitional entropy and conditional probability would effectively detect token boundaries, functionally disambiguating punctuation as well.</p><p id="__p10" class="p p-last">We evaluated our algorithm based on two problems: identification of boundaries between words and disambiguation of punctuation via tokenization. To evaluate the former, we compared our algorithm to a whitespace tokenizer. Although whitespace tokenization does not provide us with a perfect gold standard, it is very effective at finding boundaries between words, breaking only on multi-word tokens. We evaluated our algorithm&#x02019;s ability to separate punctuation from words using a manually created gold standard.</p></div><div id="__sec2" class="tsec sec"><h2 class="head no_bottom_margin" id="__sec2title">Background</h2><p id="__p11" class="p p-first">Zellig Harris<sup><a href="#b11-amia-0811-s2007" rid="b11-amia-0811-s2007" class=" bibr popnode">11</a></sup> first developed and established the utility of segmentation based on character entropy in 1967, although he used it to analyze words for their component morphemes. Using an English dictionary, Harris established the conditional probabilities of all occurring transitions between characters. He then established both the number of probable next neighbors and the reduction of entropy given each consecutive character, judging the latter to more accurately predict morpheme boundaries.</p><p id="__p12">In 2001, John Goldsmith<sup><a href="#b12-amia-0811-s2007" rid="b12-amia-0811-s2007" class=" bibr popnode">12</a></sup> built on Harris&#x02019;s work, choosing to use Minimum Description Length for the unsupervised acquisition of morphology of a natural language. He was able to achieve a precision of 85.9% and recall of 90.4% on a set of 1,000 English words.</p><p id="__p13">More recently, Shamim Ara Mollah and Stephen Johnson<sup><a href="#b13-amia-0811-s2007" rid="b13-amia-0811-s2007" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_327591658">13</a></sup> reported on the automated learning of the morphology of a medical language. After initially using Harris&#x02019;s algorithm, producing a set of probable morphemes, they pruned these morphemes based on the Mutual Information (MI) principle of Information Theory. On a set of 1,000 manually annotated medical words (generally far more complex than general English), they were able to achieve greater than 80% sensitivity and 90% specificity.</p><p id="__p14" class="p p-last">We expand on the aforementioned algorithms by applying them at a higher level, to the detection of token boundaries within texts, as opposed to the detection of morpheme boundaries within words.</p></div><div id="__sec3" class="tsec sec"><h2 class="head no_bottom_margin" id="__sec3title">Methods</h2><p id="__p15" class="p p-first">We gathered a training set of 15,082 clinician-entered free text notes written over a three year period on WebCIS, a clinical information system used in both inpatient and outpatient settings at New York Presbyterian Hospital. Notes were, on average, approximately 160 words long and were of a variety of types, such as inpatient admission, transfer and progress notes and outpatient notes.</p><p id="__p16">We then trained two <strong>tries</strong> (<a href="/pmc/articles/PMC2655800/figure/f1-amia-0811-s2007/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="f1-amia-0811-s2007" rid-ob="ob-f1-amia-0811-s2007" co-legend-rid="lgnd_f1-amia-0811-s2007"><span>Figure 1</span></a>) of arbitrarily large depth on each corpus, one reading forward and one reading backward. A trie is an ordered tree used to store associative arrays with strings as keys. The root node is associated with the empty string, and each edge is associated with a concatenated character. All descendants of the root represent a unique substring of the corpus and contain its total number of observed occurrences. We chose the trie data structure given its simplicity and its capacity to hold the number of occurrences of any substring of length less than or equal to the trie&#x02019;s depth.</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="f1-amia-0811-s2007" co-legend-rid="lgnd_f1-amia-0811-s2007"><a class="icnblk_img figpopup" href="/pmc/articles/PMC2655800/figure/f1-amia-0811-s2007/" target="figure" rid-figpopup="f1-amia-0811-s2007" rid-ob="ob-f1-amia-0811-s2007"><img src="/pmc/articles/PMC2655800/bin/amia-0811-s2007f1.gif" class="small-thumb" alt="Figure 1" title="Figure 1" src-large="/pmc/articles/PMC2655800/bin/amia-0811-s2007f1.jpg" /></a><div class="icnblk_cntnt" id="lgnd_f1-amia-0811-s2007"><div><a class="figpopup" href="/pmc/articles/PMC2655800/figure/f1-amia-0811-s2007/" target="figure" rid-figpopup="f1-amia-0811-s2007" rid-ob="ob-f1-amia-0811-s2007">Figure 1</a></div><!--caption a4--><div><span>Trie data structure. The probability of observing an &#x02018;s&#x02019; given the preceding string &#x0201c;Pt&#x0201d; is &#x000bc;, or 25%. The freedom following &#x0201c;pt&#x0201d; is 3.</span></div></div></div><p id="__p17">To calculate conditional probability of observing a character given a preceding string, we divided the observed occurrences of the concatenation of the string and the character by the observed occurrences of the string by itself. We define <strong>freedom at character transitions</strong> as the size of the set of characters that were observed to follow a substring of the corpus. To calculate freedom, we counted the children of the trie node representing the string (<a href="/pmc/articles/PMC2655800/figure/f1-amia-0811-s2007/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="f1-amia-0811-s2007" rid-ob="ob-f1-amia-0811-s2007" co-legend-rid="lgnd_f1-amia-0811-s2007"><span>Figure 1</span></a>).</p><p id="__p18">We analyzed text for specific statistical properties indicative of token boundaries. First, we calculated the forward- and backward-reading freedom of transitions following each substring of the text, lengths one through seven. We chose not to calculate the freedom substrings with a length of greater than seven because it was clear from previous work that the algorithm performed optimally using strings of length three through five, as is discussed later in the paper. We then used these values to assign additional statistical properties to characters, including increase and decrease in freedom. We assigned a &#x0201c;peak&#x0201d; value to each character transition, computed by adding the value of the preceding increase in freedom to the following decrease in freedom. We characterized token boundaries based on the sum of their forward- and backward-reading peak values.</p><p id="__p19">To maximize effectiveness of the algorithm, we determined optimal values for substring length and the threshold above which summed peak values would indicate token boundaries. First, we randomly selected a test set of twenty clinician-entered free text notes. We then tokenized the notes in our test set using different combinations of these two values. We compared our results to those of a simple whitespace tokenizer, which identifies a boundary at every space, tab or line breaking character. We defined true positives as character transitions that were classified as token boundaries in both sets of results. False positives included instances of boundary detection by the freedom-based algorithm that were not identified by the whitespace tokenizer. True negatives included all transitions not identified by either method, and false negatives included instances of boundary detection by the whitespace algorithm that were not identified by the freedom-based algorithm.</p><p id="__p20">For substring lengths three through six, we calculated average precision, recall (sensitivity), specificity and accuracy over a variable threshold. We graphed precision versus recall (PR), as well as sensitivity versus 1-specificity (ROC). We calculated areas under each curve, and used these statistics to compare effectiveness of different trie depths. Average precision, recall, specificity and accuracy served to compare effectiveness of threshold values, and the tradeoffs associated with each.</p><p id="__p21">Having identified effective trie depth and threshold values, we used these values to tokenize the test set again. We then read through each automatically identified token individually, assigning probabilities to each character given all preceding characters in the token. To break sentence ending punctuation from the end of tokens, we identified new boundaries preceding characters that had a very low probability of occurrence within these tokens.</p><p id="__p22" class="p p-last">We asked two independent, outside raters to analyze sentence-ending punctuation from our test set of twenty notes. They were asked to mark each punctuation character as an independent token or as part of a larger token. Inter-rater reliability was measured using Cohen&#x02019;s kappa. All disagreements were subsequently resolved by discussion. We then compared our algorithm&#x02019;s results to the standard developed by our raters.</p></div><div id="__sec4" class="tsec sec"><h2 class="head no_bottom_margin" id="__sec4title">Results</h2><p id="__p23" class="p p-first">Using substrings of length 3 to calculate conditional probability and freedom at character transitions, we were able to achieve a sensitivity of 99% (specificity 92%) with the threshold above which summed peak values would identify token boundaries set to 15. We achieved specificity of 99% (sensitivity 72%) setting the threshold to 40. Using substrings of length 4, we achieved sensitivity of 99% (specificity 93%) with a threshold of 25, and sensitivity of 86% (specificity 99%) with a threshold of 50. With substrings of length 5, we achieved sensitivity of 98% (specificity 94%) with a threshold of 20, and sensitivity of 88% (specificity 99%) with a threshold of 45. Using substrings of length 6, we achieved sensitivity of 98% (specificity 91%) with a threshold of 10, and sensitivity of 87% (specificity 99%) with a threshold of 35.</p><p id="__p24">We obtained the area under the ROC curve (AUC) for substrings of length three to six (<a href="/pmc/articles/PMC2655800/table/t1-amia-0811-s2007/" target="true" class="fig-table-link table figpopup" rid-figpopup="t1-amia-0811-s2007" rid-ob="ob-t1-amia-0811-s2007" co-legend-rid=""><span>Table 1</span></a>). The largest AUC was 0.990, achieved using substrings of length three. Substrings of length four, five and six achieved AUC of 0.988, 0.987 and 0.985, respectively.</p><!--table ft1--><!--table-wrap mode=article t1--><div class="table-wrap iconblock ten_col whole_rhythm clearfix" id="t1-amia-0811-s2007"><a href="/pmc/articles/PMC2655800/table/t1-amia-0811-s2007/" target="table" rid-ob="ob-t1-amia-0811-s2007" rid-figpopup="t1-amia-0811-s2007" class="table img_link icnblk_img figpopup"><img alt="Table 1" title="Table 1" class="small-thumb" src="/pmc/articles/PMC2655800/table/t1-amia-0811-s2007/?report=thumb" src-large="/pmc/articles/PMC2655800/table/t1-amia-0811-s2007/?report=previmg" /></a><div class="icnblk_cntnt"><div><a class="figpopup" href="/pmc/articles/PMC2655800/table/t1-amia-0811-s2007/" target="table" rid-figpopup="t1-amia-0811-s2007" rid-ob="ob-t1-amia-0811-s2007">Table 1</a></div><!--caption a4--><div><span>Areas under ROC and PR curves achieved using substrings of given lengths.</span></div></div></div><p id="__p25">We then obtained the area under each PR curve (AUPRC) (<a href="/pmc/articles/PMC2655800/table/t1-amia-0811-s2007/" target="true" class="fig-table-link table figpopup" rid-figpopup="t1-amia-0811-s2007" rid-ob="ob-t1-amia-0811-s2007" co-legend-rid=""><span>Table 1</span></a>). The largest AUPRC, 0.974, was achieved using substrings of length four. Substrings of length five, six and three achieved AUPRC of 0.970, 0.966 and 0.945, respectively.</p><p id="__p26" class="p p-last">When asked to classify punctuation characters as either independent or part of a larger token, our raters found 468 punctuation characters. They agreed on the classification of 452 out of 468, or 96.6% (kappa = .916). Twelve of the sixteen disagreements involved classifications of question marks that preceded dependent words or phrases. The raters classified 139, or 29.7% of all punctuation characters as part of a larger token, and the remaining 329, or 70.3% as independent. Our algorithm correctly classified 106 out of 139, or 76% of dependent punctuation characters and 271 out of 329, or 82% of independent punctuation for a total of 81% of punctuation characters classified correctly with 89.1% precision 82.0% recall.</p></div><div id="__sec5" class="tsec sec"><h2 class="head no_bottom_margin" id="__sec5title">Discussion</h2><p id="__p27" class="p p-first">Comparison between our algorithm and a simple whitespace tokenizer allowed us to determine the optimal substring length for tokenization. This is integral to the algorithm, as it must recognize common statistical patterns while simultaneously maximizing the length of patterns and their probability of occurring in the corpus. Although the tokenizer performed well with substrings of lower length in terms of AUC, precision is more important to tokenization than is specificity. Tokenizers must minimize incorrect boundary identifications, and precision measures this more accurately. Therefore, we gave more weight to the AUPRC, a reflection of precision and recall. When we used substrings of length four, the tokenizer clearly demonstrated higher performance as compared to when we used substrings of different lengths.</p><p id="__p28">Also, high levels of agreement between our own algorithm and the whitespace tokenization algorithm suggested that our algorithm discovered the function of whitespace in the corpus. This is surprising, as our algorithm was not given any knowledge of the difference between whitespace characters and any other character.</p><p id="__p29">When charged with identification of sentence-ending punctuation, our algorithm achieved precision and recall of 89.1% and 82.0%, respectively. Such performance without any knowledge of the function of punctuation around digits or the case of preceding and subsequent letters is also notable.</p><p id="__p30">Rule-based tokenizers may be handcrafted to achieve arbitrarily high effectiveness. However, an important shortcoming of rule-based tokenizers is that they are time consuming to create and maintain. Each rule must be discovered, written and tested iteratively, and rules must not conflict with each other. Furthermore, any change in the corpus, such as the use of new tokens, may require correction of existing rules or creation of new rules to tokenize properly. With an unsupervised approach, such maintenance can be fully automated.</p><p id="__p31">Supervised machine learning approaches require a set of manually tokenized documents on which to train. Expansion to a new distinct corpus involves the generation of a new training set, sacrificing portability. Although unsupervised machine learning techniques exist, they have been based on assumptions that characters such as whitespace, punctuation, digits and upper and lowercase letters fall into separate classes<sup><a href="#b14-amia-0811-s2007" rid="b14-amia-0811-s2007" class=" bibr popnode">14</a></sup>. Our technique makes no such assumptions, and is therefore easily applied to corpora that regularly violate these assumptions.</p><p id="__p32">The most important limitation of our work was the lack of gold standard against which we could evaluate our tokenization algorithm. Historically, tokenization is not well defined on purely linguistic or formal grounds, and success is dependent mostly on the underlying application<sup><a href="#b15-amia-0811-s2007" rid="b15-amia-0811-s2007" class=" bibr popnode">15</a></sup>. The creation of a gold standard is very costly in terms of time and effort, and a single gold standard will not serve to effectively evaluate tokenization that is to be used for different purposes, and in different NLP systems.</p><p id="__p33" class="p p-last">Future studies should include comparison with a broad, manually generated gold standard. Such a gold standard should contain multi-word phrases judged to be semantically linked, justifying corresponding tokenization. Future studies should also include evaluation our tokenization algorithm with respect to its utility within a statistical NLP system. The characteristics and potential contribution of each of the many features assigned to character transitions by our algorithm may also contribute to our approach. There exists a deep relationship between tokenization and data compression in information theory that should also be studied in an effort to define tokenization in terms of its statistical properties.</p></div><div id="__sec6" class="tsec sec"><h2 class="head no_bottom_margin" id="__sec6title">Conclusions</h2><p id="__p34" class="p p-first-last">The dynamic nature of biomedical language and the existence of many distinct sub-languages necessitate a general approach to tokenization. Our algorithm, effective at token boundary detection and punctuation disambiguation, is easily trained, offering excellent portability between corpora and adaptability to new user behavior.</p></div><div id="__ackidm140522521603952" class="tsec sec"><h2 class="head no_bottom_margin" id="__ackidm140522521603952title">Acknowledgments</h2><div class="sec"><p id="__p35">Jesse Wrenn is supported by National Library of Medicine Informatics Training Grant number T15-LM07079, as well as by Columbia University&#x02019;s Department of Biomedical Informatics. Stephen Johnson&#x02019;s project entitled &#x0201c;Using Narrative Data to Enrich the Online Medical Record&#x0201d; is supported by grant number 5R01 LM007268-03. Peter Stetson is supported by National Library of Medicine grant K22LM008805.</p></div></div><div id="__ref-listidm140522521602880" class="tsec sec"><h2 class="head no_bottom_margin" id="__ref-listidm140522521602880title">References</h2><div class="ref-list-sec sec" id="reference-list"><div class="ref-cit-blk half_rhythm" id="b1-amia-0811-s2007">1. <span class="element-citation">Hripcsak G, Friedman C, Alderson PO, DuMouchel W, Johnson SB, Clayton PD. Unlocking clinical data from narrative reports: a study of natural language processing. <span><span class="ref-journal">Ann. Intern. Med. </span>1995;<span class="ref-vol">122</span>:681–8.</span>  <span class="nowrap ref pubmed">[<a href="/pubmed/7702231" target="pmc_ext" ref="reftype=pubmed&amp;article-id=2655800&amp;issue-id=177326&amp;journal-id=362&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="b2-amia-0811-s2007">2. <span class="element-citation">Friedman C et al. GENIES: A natural language processing system for the extraction of molecular pathways from journal articles. <span><span class="ref-journal">Bioinformatics. </span>2001;<span class="ref-vol">17</span>:S74–82.</span>  <span class="nowrap ref pubmed">[<a href="/pubmed/11472995" target="pmc_ext" ref="reftype=pubmed&amp;article-id=2655800&amp;issue-id=177326&amp;journal-id=362&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="b3-amia-0811-s2007">3. <span class="element-citation">Institute of Medicine  . <span class="ref-journal">Performance Measurement: Accelerating Improvement.</span> Washington, DC: National Academy Pr; 2006. </span></div><div class="ref-cit-blk half_rhythm" id="b4-amia-0811-s2007">4. <span class="element-citation">Persell SD, Wright JM, Thompson JA, Kmetik KS, Baker DW. Assessing the Validity of National Health Quality Measures for Coronary Artery Disease Using an Electronic Health Record. <span><span class="ref-journal">Archives of Internal Medicine. </span>2006;<span class="ref-vol">166</span>:2272–77.</span>  <span class="nowrap ref pubmed">[<a href="/pubmed/17101947" target="pmc_ext" ref="reftype=pubmed&amp;article-id=2655800&amp;issue-id=177326&amp;journal-id=362&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="b5-amia-0811-s2007">5. <span class="element-citation">Chen L, Friedman C. &#x02018;Extracting phenotypic information from the literature via natural language processing&#x02019;. &#x02018;Proceedings of the 11th World Congress on Medical Informatics&#x02019;; IMIA, San Francisco, CA. 2004. pp. 758–762.</span></div><div class="ref-cit-blk half_rhythm" id="b6-amia-0811-s2007">6. <span class="element-citation">Webster Jonathan J, Chunyu Kit. Tokenization as the initial phase in NLP. Proceedings of the 14th International Conference on Computational Linguistics (COLING'92); Nantes, France. 1992. pp. 1pp. 106–1.pp. 110</span></div><div class="ref-cit-blk half_rhythm" id="b7-amia-0811-s2007">7. <span class="element-citation">Grefenstette Gregory, Tapanainen Papi. What is a word, What is a sentence? Problems of Tokenization. Proceedings of the 3rd Conference on Computational Lexicography and Text Research (COMPLEX &#x02019;94; 1994. </span></div><div class="ref-cit-blk half_rhythm" id="b8-amia-0811-s2007">8. <span class="element-citation">Mikheev Andre. Periods, Capitalized Words, etc. <span><span class="ref-journal">Computational Linguistics. </span>2002;<span class="ref-vol">28</span>(3):289–318.</span></span></div><div class="ref-cit-blk half_rhythm" id="b9-amia-0811-s2007">9. <span class="element-citation">Stetson Peter D, Johnson Stephen B, Scotch Matthew, Hripcsak George. The Sublanguage of Cross-Coverage. <span><span class="ref-journal">AMIA 2002 Symposium Proceedings. </span>2002;<span class="ref-vol">742</span></span> <span class="nowrap ref pmc">[<a class="int-reflink" href="/pmc/articles/PMC2244148/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="/pubmed/12463923" target="pmc_ext" ref="reftype=pubmed&amp;article-id=2655800&amp;issue-id=177326&amp;journal-id=362&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="b10-amia-0811-s2007">10. <span class="element-citation">Barrows RC, Jr, Busuioc M, Friedman C. Limited parsing of notational text visit notes: ad-hoc vs. NLP approaches. <span><span class="ref-journal">Proc AMIA Symp. </span>2000;<span class="ref-vol">7</span>(6):51–5.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="/pmc/articles/PMC2243829/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="/pubmed/11079843" target="pmc_ext" ref="reftype=pubmed&amp;article-id=2655800&amp;issue-id=177326&amp;journal-id=362&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="b11-amia-0811-s2007">11. <span class="element-citation">Harris Zellig. Morpheme boundaries within words: Report on a computer test. <span><span class="ref-journal">Transformations and Discourse Analysis Papers. </span>1967;<span class="ref-vol">73</span></span></span></div><div class="ref-cit-blk half_rhythm" id="b12-amia-0811-s2007">12. <span class="element-citation">Goldsmith John. Unsupervised Learning of the Morphology of a Natural Language. <span><span class="ref-journal">Computational Linguistics. </span>2001;<span class="ref-vol">27</span>(2):153–198.</span></span></div><div class="ref-cit-blk half_rhythm" id="b13-amia-0811-s2007">13. <span class="element-citation">Mollah Shamim Ara, Johnson Stephen B. Automatic Learning of the Morphology of Medical Language using Information Compression. <span><span class="ref-journal">AMIA 2003 Symposium Proceedings. </span>2003;<span class="ref-vol">938</span></span> <span class="nowrap ref pmc">[<a class="int-reflink" href="/pmc/articles/PMC1480252/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="/pubmed/14728443" target="pmc_ext" ref="reftype=pubmed&amp;article-id=2655800&amp;issue-id=177326&amp;journal-id=362&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="b14-amia-0811-s2007">14. <span class="element-citation">Schmid H. Unsupervised Learning of Period Disambiguation for Tokenisation. Internal Report, IMS, University of Stuttgart.2000. </span></div><div class="ref-cit-blk half_rhythm" id="b15-amia-0811-s2007">15. <span class="element-citation">Habert B, Adda G, Adda-Decker M, Boula de Mar&#x000eb;uil P, Ferrari S, Ferret O, Illouz G, Paroubek P. Towards Tokenization Evaluation. Proc. First International Conference on Language Resources and Evaluation (LREC); Granada, Spain. May, 1998. pp. 427–431.</span></div><div class="ref-cit-blk half_rhythm" id="b16-amia-0811-s2007">16. <span class="element-citation">Mikheev Andre. Document Centered Approach to Text Normalization. Proceedings of the Annual ACM Conference on Research and Development in Information Retrieval (SIGIR &#x02019;00); 2000. pp. 136–143.</span></div></div></div></div><!--post-content--><hr class="whole_rhythm no_bottom_margin" /><div class="courtesy-note no_margin small">Articles from <span class="acknowledgment-journal-title">AMIA Annual Symposium Proceedings</span> are provided here courtesy of <strong>American Medical Informatics Association</strong></div></div>
            
            
        
            
        </div>
        <!-- Book content -->
    </div>
    
    <div id="rightcolumn" class="four_col col last">
        <!-- Custom content above discovery portlets -->
        <div class="col6">
            
        </div>
        
        <div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><div class="format-menu"><h2>Formats:</h2><ul><li class="selected">Article</li> | <li><a href="/pmc/articles/PMC2655800/?report=reader">PubReader</a></li> | <li class="epub-link"><a href="/pmc/articles/PMC2655800/epub/">ePub (beta)</a></li> | <li><a href="/pmc/articles/PMC2655800/pdf/amia-0811-s2007.pdf">PDF (106K)</a></li> | <li><a href="#" data-citationid="PMC2655800" class="citationexporter ctxp">Citation</a></li></ul></div></div><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="share-buttons"><h2>Share</h2><ul><li class="facebook"><a href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2655800%2F"><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4158155/img/4047626" alt="Share on Facebook" />
                             Facebook
                        </a></li><li class="twitter"><a href="https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2655800%2F&amp;text=An%20Unsupervised%20Machine%20Learning%20Approach%20to%20Segmentation%20of%20Clinician-Entered%20Free%20Text"><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4158155/img/4047627" alt="Share on Twitter" />
                             Twitter
                        </a></li><li class="gplus"><a href="https://plus.google.com/share?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2655800%2F"><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4158155/img/4047628" alt="Share on Google Plus" />
                             Google+
                        </a></li></ul></div>
        
        <div id="ajax-portlets" data-pmid="18693949" data-aiid="2655800" data-aid="2655800" data-iid="177326" data-domainid="362" data-domain="procamia" data-accid="PMC2655800" data-md5="c3d747f0d61891730fcaea35b4c83861"></div>
                
        <!-- Custom content below discovery portlets -->
        <div class="col7">
            
        </div>
    </div>
</div>

<!-- Custom content after all -->
<div class="col8">
    
</div>
<div class="col9">
    
</div>

<script src="/corehtml/pmc/js/jquery.scrollTo-1.4.2.js"></script>
<script>
    (function($){
        $('.skiplink').each(function(i, item){
            var href = $($(item).attr('href'));
            href.attr('tabindex', '-1').addClass('skiptarget'); // ensure the target can receive focus
            $(item).on('click', function(event){
                event.preventDefault();
                $.scrollTo(href, 0, {
                    onAfter: function(){
                        href.focus();
                    }
                });
            });
        });
    })(jQuery);
</script>



<div id="body-link-poppers"></div>
                        </div>
                        <div class="bottom">
                            
                            <div id="NCBIFooter_dynamic">
    <a id="help-desk-link" class="help_desk jig-ncbihelpwindow" target="_blank" href="">Support Center</a>
    <a id="help-desk-link" class="help_desk jig-ncbihelpwindow" target="_blank" href="https://support.ncbi.nlm.nih.gov/ics/support/KBList.asp?Time=2017-06-24T03:34:21-04:00&amp;Snapshot=%2Fprojects%2FPMC%2FPMCViewer@4.45&amp;Host=ptpmc202&amp;ncbi_phid=F4FC06DA94E13C510000000000E200B8&amp;ncbi_session=CE89646793B9E881_0084SID&amp;from=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2655800%2F&amp;Db=pmc&amp;folderID=132&amp;Ncbi_App=pmc&amp;Page=literature&amp;style=classic&amp;deptID=28049">Support Center</a>
    
</div>

                            <div class="footer" id="footer">
    
    <div class="subfooter"> </div><script type="text/javascript" src="/portal/portal3rc.fcgi/static/js/preloaderWidget.js"> </script>
    <div id="external-disclaimer" class="offscreen_noflow">
        External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.
    </div>    
    <div id="ncbifooter" class="contact_info">      
        <div id="footer-contents-right">
            <div id="nlm_thumb_logo">
                <a href="https://www.nlm.nih.gov" title="NLM">NLM</a>
            </div>
            <div id="nih_thumb_logo">
                <a href="https://www.nih.gov" title="NIH">NIH</a>
            </div>
            <div id="hhs_thumb_logo">
                <a href="https://www.hhs.gov" title="DHHS">DHHS</a>
            </div>
            <div id="usagov_thumb_logo">
                <a href="https://www.usa.gov" title="USA.gov">USA.gov</a>
            </div>         
        </div>
        
        <div id="footer-contents-left">
            <p class="address vcard">
                <span class="url">
                    <a class="fn url newdomain" href="https://www.ncbi.nlm.nih.gov">National Center for
                        Biotechnology Information</a>,
                </span> <span class="org url newdomain"><a href="https://www.nlm.nih.gov/">U.S. National Library of Medicine</a></span>
                <span class="adr">
                    <span class="street-address">8600 Rockville Pike</span>, <span class="locality">Bethesda</span>
                    <span class="region">MD</span>, <span class="postal-code">20894</span>
                    <span class="country-name">USA</span>
                </span>
            </p>
            
            <a href="/home/about/policies.shtml">Policies and Guidelines</a> | <a href="/home/about/contact.shtml">Contact</a>
        </div>
    </div>
    <script type="text/javascript" src="/portal/portal3rc.fcgi/rlib/js/InstrumentOmnitureBaseJS/InstrumentNCBIConfigJS/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js?v=1"> </script>    
    <script type="text/javascript" src="/portal/portal3rc.fcgi/static/js/hfjs2.js"> </script>
</div>
                        </div>
                    </div>
                    <!--/.page-->
                </div>
                <!--/.wrap-->
            </div><!-- /.twelve_col -->
        </div>
        <!-- /.grid -->

        <span class="PAFAppResources"></span>
        
        <!-- BESelector tab -->
        
        
        
        <noscript><img alt="statistics" src="/stat?jsdisabled=true&amp;ncbi_db=pmc&amp;ncbi_pdid=article&amp;ncbi_acc=&amp;ncbi_domain=procamia&amp;ncbi_report=record&amp;ncbi_type=fulltext&amp;ncbi_objectid=&amp;ncbi_pcid=/articles/PMC2655800/&amp;ncbi_app=pmc" /></noscript>
        
        
        <!-- usually for JS scripts at page bottom -->
        <!--<component id="PageFixtures" label="styles"></component>-->
    

<!-- CE89646793B9E881_0084SID /projects/PMC/PMCViewer@4.45 ptpmc202 v4.1.r526566 Fri, Feb 03 2017 14:27:04 -->

<script type="text/javascript" src="//static.pubmed.gov/portal/portal3rc.fcgi/4158155/js/3879255/4121861/3818874/3821238/4117325/4087685/4072593/4076480/3921943/4105668/4065628.js" snapshot="pmc"></script></body>
</html>